src/components/repetitions/Overview.vue

```diff
@@ -101,64 +103,53 @@ export default {
     Badge,
     PopupDialog
   },
-  data() {
-    const projectId = this.$route.params.projectId;
-    const eventId = this.$route.params.eventId;
-
+  data: function () {
     return {
-      projectId,
-      eventId,
-      actualEvent: {},
-      groupedRepetitions: []
+      groupedRepetitions: new Map()
     };
   },
   computed: {
-    /**
-     * @return {GroupedEvent}
-     */
-    event() {
-      return this.$store.getters.getProjectEventById(this.projectId, this.eventId);
+    eventId() {
+      return this.$route.params.eventId;
+    },
+    projectId() {
+      return this.$route.params.projectId;
     },
-
     since() {
       const now = new Date();
-      const firstOccurence = new Date(this.event.payload.timestamp);
-      const differenceInDays = (now - firstOccurence) / (1000 * 3600 * 24);
+      const firstOccurrence = new Date(this.event.payload.timestamp);
+      const differenceInDays = (now - firstOccurrence) / (1000 * 3600 * 24);
 
       return `${Math.round(differenceInDays)} days`;
     }
   },
   async created() {
-    /**
-     * actual event is original event merged with latest repetitions
-     *
-     * For that we use GET_LATEST_EVENT action that merges event from store with repetition
-     * @type {GroupedEvent}
-     */
-    this.actualEvent = await this.$store.dispatch(GET_LATEST_EVENT, { projectId: this.projectId, eventId: this.eventId });
+    this.event = this.$store.getters.getProjectEventById(this.projectId, this.eventId);
+
+    if (!this.event || this.event.payload) {
+      this.event = await this.$store.dispatch(FETCH_EVENT_REPETITION, {
+        projectId: this.projectId,
+        eventId: this.eventId
+      });
+    }
 
     /**
      * Dispatching action that fetches several latest repetitions
-     * @type {GroupedEvent[]}
      */
     const repetitions = await this.$store.dispatch(FETCH_EVENT_REPETITIONS, {
       projectId: this.projectId,
-      eventId: this.eventId
+      eventId: this.eventId,
+      limit: 10
     });
 
-    console.log('repetitions', repetitions);
-
     /**
      * We use Map here to save the key's order,
      * `Object` does not guarantee the iteration order
-     * @type {Map<String, GroupedEvent[]>}
+     * @type {Map<String, HawkEvent[]>}
```

Comment:
this is not a HawkEvent

<img width="466" alt="image" src="https://user-images.githubusercontent.com/3684889/74462228-f0ce3080-4ea0-11ea-8396-137546e758ec.png">